### PR TITLE
Remove CreateIfMissing option

### PR DESCRIPTION
**Pull request by chemelnucfin**
Partial #4111

### CODE REVIEW

File: firestore/google/cloud/firestore_v1beta1/_helpers.py [hunk 1]
```diff
@@ -897,7 +893,7 @@ def pb_for_delete(document_path, option):
     """
     write_pb = write_pb2.Write(delete=document_path)
     if option is not None:
-        option.modify_write(write_pb, no_create_msg=NO_CREATE_ON_DELETE)
+        option.modify_write(write_pb)
```
Comment (dhermes) [inline firestore/google/cloud/firestore_v1beta1/_helpers.py]: @chemelnucfin Why was this removed but not replaced with something equivalent?

File: firestore/google/cloud/firestore_v1beta1/client.py [hunk 2]
```diff
@@ -252,10 +251,8 @@ def write_option(**kwargs):
         :meth:`~.DocumentReference.update` and
         :meth:`~.DocumentReference.delete`.
 
-        Exactly one of three keyword arguments must be provided:
+        One of the following two keyword arguments must be provided:
```
Comment (dhermes) [inline firestore/google/cloud/firestore_v1beta1/client.py]: Can you just remove `two` here? (`three` should've been removed from the previous description)

Reasoning: the count is clear from the bullets but is just one more thing we have to "maintain"

File: firestore/tests/unit/test__helpers.py [hunk 3]
```diff
@@ -1216,6 +1217,9 @@ def _helper(self, option=None, do_transform=False, **write_kwargs):
         map_pb = document_pb2.MapValue(fields={
             'yum': _value_pb(bytes_value=value),
         })
+        if isinstance(option, ExistsOption):
+            write_kwargs = {'current_document' : {'exists': option._exists}}
```
Comment (dhermes) [inline firestore/tests/unit/test__helpers.py]: @chemelnucfin 

1. This should update ``write_kwargs`` but should not discard the previous values.
1. The value should be a protobuf, not a dictionary

File: firestore/google/cloud/firestore_v1beta1/_helpers.py (hunk 1 above)
Comment (chemelnucfin) [reply firestore/google/cloud/firestore_v1beta1/_helpers.py]: That message is only good for create_if_missing.

File: firestore/tests/unit/test__helpers.py (hunk 3 above)
Comment (chemelnucfin) [reply firestore/tests/unit/test__helpers.py]: would an update work on write_args?

the value gets used in the protobuf below I think?

**Comment by chemelnucfin** [comment]
Closing this pull request

The commits are in #4851 and can be reviewed commit by commit.
